src/main/java/frc/robot/subsystems/hood/Hood.java

```diff
@@ -0,0 +1,98 @@
+package frc.robot.subsystems.hood;
+
+import edu.wpi.first.units.Angle;
+import edu.wpi.first.units.MutableMeasure;
+import edu.wpi.first.units.Units;
+import edu.wpi.first.wpilibj.smartdashboard.Mechanism2d;
+import edu.wpi.first.wpilibj.smartdashboard.MechanismLigament2d;
+import edu.wpi.first.wpilibj.smartdashboard.MechanismRoot2d;
+import edu.wpi.first.wpilibj2.command.Command;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import java.util.function.Supplier;
+import lib.Utils;
+import org.littletonrobotics.junction.AutoLogOutput;
+import org.littletonrobotics.junction.Logger;
+
+public class Hood extends SubsystemBase {
+    private static Hood INSTANCE = null;
+    private final HoodInputsAutoLogged inputs = HoodIO.inputs;
+    private final HoodIO io;
+    @AutoLogOutput private final Mechanism2d mechanism2d = new Mechanism2d(3, 3);
+    private final MechanismRoot2d root =
+            mechanism2d.getRoot(
+                    "Hood",
+                    HoodConstants.mechanism2dPose.getX(),
+                    HoodConstants.mechanism2dPose.getY());
+    private final MechanismLigament2d hood =
+            root.append(new MechanismLigament2d("Hood", HoodConstants.HoodLength, 45));
+
+    private ControlMode controlMode = ControlMode.POSITION;
+
+    /**
+     * Constructor for Hood subsystem.
+     *
+     * @param io IO of the subsystem.
+     */
+    private Hood(HoodIO io) {
+        this.io = io;
+    }
+
+    /**
+     * Gets the single instance of hoodSubsystem.
+     *
+     * @return The single instance of hoodSubsystem.
+     */
+    public static Hood getInstance() {
+        return INSTANCE;
+    }
+
+    public static void initialize(HoodIO io) {
+        INSTANCE = new Hood(io);
+    }
+
+    public MutableMeasure<Angle> getAngle() {
+        return inputs.angle;
+    }
+
+    public boolean atSetpoint() {
+        return Utils.epsilonEquals(inputs.powerSetpoint * 12, inputs.voltage.in(Units.Volts));
```

Comment:
Can you explain the usecase of this function?

Reply:
So we know when to stop the setAngle command

Reply:
I meant for this function to check whether the hood is at its positional setpoint. No point in checking the voltage setpoint.